service/collector/resource_group.go

```diff
@@ -67,17 +72,26 @@ func NewResourceGroup(config ResourceGroupConfig) (*ResourceGroup, error) {
 		k8sClient: config.K8sClient,
 		logger:    config.Logger,
 
-		environmentName: config.EnvironmentName,
+		environmentName:        config.EnvironmentName,
+		cpAzureClientSetConfig: config.CPAzureClientSetConfig,
 	}
 
 	return r, nil
 }
 
 func (r *ResourceGroup) Collect(ch chan<- prometheus.Metric) error {
-	clientSets, err := getClientSets(r.k8sClient, r.environmentName)
+	clientSets, err := credential.GetAzureClientSetsFromCredentialSecretsBySubscription(r.k8sClient, r.environmentName)
```

Comment:
This collector will still expose metrics only by subscription. Resource groups are the same even using different client id or tenant id.